modules/auxiliary/scanner/http/vicidial_multiple_sqli.rb

```diff
@@ -0,0 +1,300 @@
+      'ready_max_logout' => '-1',
+      'status_group_id' => '',
+      'custom_one' => '',
+      'custom_two' => '',
+      'custom_three' => '',
+      'custom_four' => '',
+      'custom_five' => '',
+      'qc_enabled' => '0',
+      'qc_user_level' => '1',
+      'qc_pass' => '0',
+      'qc_finish' => '0',
+      'qc_commit' => '0',
+      'realtime_block_user_info' => '0',
+      'admin_hide_lead_data' => '0',
+      'admin_hide_phone_data' => '0',
+      'ignore_group_on_search' => '0',
+      'user_admin_redirect_url' => '',
+      'view_reports' => '0',
+      'access_recordings' => '0',
+      'alter_agent_interface_options' => '0',
+      'modify_users' => '0',
+      'change_agent_campaign' => '0',
+      'delete_users' => '0',
+      'modify_usergroups' => '0',
+      'delete_user_groups' => '0',
+      'modify_lists' => '0',
+      'delete_lists' => '0',
+      'load_leads' => '0',
+      'modify_leads' => '0',
+      'export_gdpr_leads' => '0',
+      'download_lists' => '0',
+      'export_reports' => '0',
+      'delete_from_dnc' => '0',
+      'modify_campaigns' => '0',
+      'campaign_detail' => '0',
+      'delete_campaigns' => '0',
+      'modify_ingroups' => '0',
+      'delete_ingroups' => '0',
+      'modify_inbound_dids' => '0',
+      'delete_inbound_dids' => '0',
+      'modify_custom_dialplans' => '0',
+      'modify_remoteagents' => '0',
+      'delete_remote_agents' => '0',
+      'modify_scripts' => '0',
+      'delete_scripts' => '0',
+      'modify_filters' => '0',
+      'delete_filters' => '0',
+      'ast_admin_access' => '0',
+      'ast_delete_phones' => '0',
+      'modify_call_times' => '0',
+      'delete_call_times' => '0',
+      'modify_servers' => '0',
+      'modify_shifts' => '0',
+      'modify_phones' => '0',
+      'modify_carriers' => '0',
+      'modify_email_accounts' => '0',
+      'vKik' => 'vKik',
+      'modify_labels' => '0',
+      'modify_colors' => '0',
+      'modify_languages' => '0',
+      'modify_statuses' => '0',
+      'modify_voicemail' => '0',
+      'modify_audiostore' => '0',
+      'modify_moh' => '0',
+      'modify_tts' => '0',
+      'modify_contacts' => '0',
+      'callcard_admin' => '0',
+      'modify_auto_reports' => '0',
+      'add_timeclock_log' => '0',
+      'modify_timeclock_log' => '0',
+      'delete_timeclock_log' => '0',
+      'manager_shift_enforcement_override' => '0',
+      'pause_code_approval' => '0',
+      'admin_cf_show_hidden' => '0',
+      'modify_ip_lists' => '0',
+      'ignore_ip_list' => '0',
+      'two_factor_override' => 'NOT_ACTIVE',
+      'vdc_agent_api_access' => '0',
+      'api_list_restrict' => '0',
+      'api_allowed_functions[]' => 'ALL_FUNCTIONS',
+      'api_only_user' => '0',
+      'modify_same_user_level' => '1',
+      'alter_admin_interface_options' => '1',
+      'SUBMIT' => 'SUBMIT'
+    }
+  end
+
+  def basic_auth
+    user_pass = datastore['USERNAME'].to_s + ':' + datastore['PASSWORD'].to_s
```

Comment:
Just curious if running `.to_s` on both of these datastore values is necessary? 

I stopped execution at the top of this method and confirmed both datastore values Strings:
```
[5] pry(#<Msf::Modules::Auxiliary__Scanner__Http__Vicidial_multiple_sqli::MetasploitModule>)> datastore['USERNAME'].class
=> String
[6] pry(#<Msf::Modules::Auxiliary__Scanner__Http__Vicidial_multiple_sqli::MetasploitModule>)> datastore['PASSWORD'].class
=> String
```

```suggestion
    user_pass = datastore['USERNAME'] + ':' + datastore['PASSWORD']
```

I'd be happy to disregard this suggestion if I'm missing something here. Thanks!

Reply:
that was copy pasta from another module, while it works, I think the best answer is to just interpolate it and not deal with `+` at all.